src/rs.cpp

```diff
@@ -1564,15 +1565,18 @@ void rs2_context_add_software_device(rs2_context* ctx, rs2_device* dev, rs2_erro
     
     auto dev_info = std::make_shared< software_device_info >( ctx->ctx );
     dev_info->set_device( std::dynamic_pointer_cast< software_device >( dev->device ) );
-    ctx->ctx->add_software_device( dev_info );
+    ctx->ctx->add_device( dev_info );
 }
 HANDLE_EXCEPTIONS_AND_RETURN(, ctx, dev)
 
 void rs2_context_remove_device(rs2_context* ctx, const char* file, rs2_error** error) BEGIN_API_CALL
 {
     VALIDATE_NOT_NULL(ctx);
     VALIDATE_NOT_NULL(file);
-    ctx->ctx->remove_device(file);
+    // The context uses the address from the device-info to maintain the list of devices. I.e., we need a device-info
+    // that uses a device-info that is_same_as() the one created above, in rs2_context_add_device:
+    auto dev_info = std::make_shared< playback_device_info >( ctx->ctx, file );
```

Comment:
Can't we find a cleaner solution for that?
This looks weird..
We will try to open a file when we wish to remove it?

Reply:
This requires a deep dive their which I currently cannot do.
So, I will just pop up it looks weird.
If you are OK with that, I trust your judgement.